### PR TITLE
import FerredoxinUI

### PR DESCRIPTION
**Pull request by singleNeuron**
# Import FerredoxinUI

## 更改内容 / Types of Changes
- UI界面 User Interface
- 内核 Core

## 检查列表 / Checklist

- [x] 我的代码符合本仓库的代码规范 My code follows the code style of this project
- [x] 我已经测试了这些更改，它们可以正常工作，既不会破坏原有任何功能(或我可以解决)，也不会影响对旧版本的支持 I have tested the changes and verified that they work and don't break anything(as well as I can manage) or drop the support for previous versions.
- [x] 我已合并了对后续工作无意义的commit并确认不会对后续维护造成困扰 I have merged commits that are meaningless for follow-up work and confirmed that they will not cause damage to follow-up maintenance

### CODE REVIEW

**Comment by qwq233** [comment]
run tests.

**Comment by qwq233** [comment]
run tests.

File: app/src/main/java/nil/nadph/qnotified/util/LicenseStatus.java [hunk 1]
```diff
@@ -65,95 +70,76 @@ public static boolean hasUserAcceptEula() {
...
-        int currentStatus = ConfigManager.getDefaultConfig()
-            .getIntOrDefault(qn_user_auth_status, -1);
+        int currentStatus = ConfigManager.getDefaultConfig().getIntOrDefault(qn_user_auth_status, -1);
         if (currentStatus == UserStatusConst.notExist) {
             LicenseStatus.setUserCurrentStatus();
-            currentStatus = ConfigManager.getDefaultConfig()
-                .getIntOrDefault(qn_user_auth_status, -1);
+            currentStatus = ConfigManager.getDefaultConfig().getIntOrDefault(qn_user_auth_status, -1);
         }
-        long lastUpdate = ConfigManager.getDefaultConfig()
```
Comment (qwq233) [inline app/src/main/java/nil/nadph/qnotified/util/LicenseStatus.java:114]: 虚空了

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.0-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.0.2-bin.zip
```
Comment (qwq233) [inline gradle/wrapper/gradle-wrapper.properties:3]: 最新版是7.1.1

**Review by qwq233 — CHANGES_REQUESTED** [review]

File: gradle/wrapper/gradle-wrapper.properties (hunk 2 above)
Comment (singleNeuron) [reply gradle/wrapper/gradle-wrapper.properties:3]: 我用的as自动升级，升到的就是702

File: settings.gradle [hunk 3]
```diff
@@ -1,10 +1,12 @@
-include ':app', ':stub', ':annotations', ':compiler', ':dex-ptm', ':mmkv'
+include ':app', ':stub', ':annotations', ':compiler', ':dex-ptm', ':mmkv' , ':FerredoxinUILib', ':FerredoxinUI:demo'
```
Comment (qwq233) [inline settings.gradle:1]:
```suggestion
include ':app', ':stub', ':annotations', ':compiler', ':dex-ptm', ':mmkv' , ':FerredoxinUILib'
```

**Review by keta1 — APPROVED** [review]

**Review by qwq233 — APPROVED** [review]

**Comment by loser641** [comment]
unsubscribe------------------&nbsp;原始邮件&nbsp;------------------
***@***.***&gt;
发送时间:&nbsp;2021年7月24日(星期六) 凌晨3:07
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;Re: [ferredoxin/QNotified] import FerredoxinUI (#974)

**Pull request merged**